internal/transport/handler_server_test.go

```diff
@@ -269,6 +269,32 @@ func (s) TestHandlerTransport_HandleStreams(t *testing.T) {
 		if want := "/service/foo.bar"; s.method != want {
 			t.Errorf("stream method = %q; want %q", s.method, want)
 		}
+
+		err := s.SetHeader(metadata.Pairs("custom-header", "Custom header value"))
```

Comment:
If desired, I can break this out into multiple test cases, but I think it's reasonable to exercise all features in the "happy path" test case as long as there is no mutual interference.

Reply:
This is fine, thanks.